os/kernel/init/os_start.c

```diff
@@ -599,6 +600,8 @@ void os_start(void)
 
 		/* Perform any processor-specific idle state operations */
 
-		up_idle();
+#ifdef CONFIG_PM
+		pm_idle();
+#endif
```

Comment:
I guess #else should be there to goto  wfi state in case there is no support for PM at all. 

#ifdef CONFIG_PM
           pm_idle();
#else 
        /* Sleep until an interrupt occurs to save power */
#if 0
        asm("WFI");                                     /* For example */
#endif

Reply:
Added this as up_cpuidle() in arch